src/com/dtmilano/android/viewclient.py

```diff
@@ -4207,6 +4218,26 @@ def excerpt(_str, execute=False):
         else:
             return code
 
+class WaitUntilViewsLoad:
+    '''
+    wait until uiAutomatorHelper findObject and click
+    '''
+    def __init__(self, uiAutomatorHelper, type=None):
+        self.uiAutomatorHelper = uiAutomatorHelper
+        self.currentDump = None
+        self.prevDump = None
+        self.type = type
+
+    def __enter__(self):
+        self.prevDump = self.uiAutomatorHelper.dumpWindowHierarchy()
+
+    def __exit__(self, exception_type, exception_value, traceback):
+        ViewClient.waitUntil(self.viewHasLoaded)
+
+    def viewHasLoaded(self):
+        self.currentDump = self.uiAutomatorHelper.dumpWindowHierarchy()
+        return self.currentDump != self.prevDump
```

Comment:
The idea behind the implementation is to delay the next touch/click until the current one is done and the promise is to get False not True. Actually, It would timeout if I get True, which is the case of no change has happened on the screen!